### PR TITLE
Start moving testing documentation into a single location

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
Start moving testing documentation into a single location
Update `test-configuration.md` with more details on creating a test and requirements

Working on #18594

### CODE REVIEW

File: Documentation/building/test-configuration.md [hunk 1]
```diff
@@ -1,41 +1,54 @@
...
- * Will not execute it. 
+## Test "Kind"
 
+* Build Only
+  * Builds an executable.
+  * Will not execute.
+  * e.g. `<CLRTestKind>BuildOnly</CLRTestKind>`
 * Run Only
-> `<CLRTestKind>RunOnly</CLRTestKind>`
+  * Can use Output of Build and Run Project with different command line arguments.
```
Comment (sbomer) [inline Documentation/building/test-configuration.md]: If I'm interpreting this correctly, I think something like "Can use output of Build or BuildAndRun projects with different command line arguments." would be clearer.

File: Documentation/building/test-configuration.md [hunk 2]
```diff
@@ -1,41 +1,54 @@
...
- * Will execute said executable. 
-* Shared Libraries
-> `<CLRTestKind>SharedLibrary</CLRTestKind>`
+## Priority
+
+Test cases are categorized by priority level. The most important subset should be and is the smallest subset. This subset is called priority 0.
 
- * For building libraries common to zero or more tests. 
+* By default, a test case is priority 0. Tests must be explicitly de-prioritized.
+* Set the priority of a test by setting the property `<CLRTestPriority>` to the test's project file.
```
Comment (sbomer) [inline Documentation/building/test-configuration.md]: **in** the test's project file.

File: Documentation/building/test-configuration.md [hunk 3]
```diff
@@ -1,41 +1,54 @@
...
+## Adding Test Guidelines
 
-By default (i.e. if not specified explicitly) a project file is BuildAndRun.
+* All test source files should include the following banner:
+```
+    // Licensed to the .NET Foundation under one or more agreements.
+    // The .NET Foundation licenses this file to you under the MIT license.
+    // See the LICENSE file in the project root for more information.
+```
+* Disable building of a test by conditionally settings the `<DisableProjectBuild>` property.
```
Comment (sbomer) [inline Documentation/building/test-configuration.md]: "setting"

**Comment by chsienki** [comment]
@sbomer  Hmm, most of it seems the same, but i'll make a note to go over and check it's still up to date when I'm done, thanks.

**Pull request merged**